scalableminds/webknossos
# Fix source maps for chrome debugging

Very annoyingly, for the last months, debugging in chrome hasn't worked for me and I think a couple others as well. This PR is an attempt to fix the source maps to solve this problem.
While researching I found out that this is a known problem (at least in chrome) and many others are affected. However in the last month the webpack developers published a new version which is supposed to fix the issue (webpack >= 2.3.2) - at least almost:
https://github.com/webpack/webpack/issues/4423#issuecomment-284872227
:p. 

I also found this handy matrix for the different (very poorly documented) source maps settings for webpack:
https://github.com/webpack/webpack/issues/2145#issuecomment-294361203
My tests somewhat confirmed his findings - at least I can indeed debug again, although I don't seem to be able to set breakpoints EVERYWHERE, but it works in most cases. Error stack traces show the bundle file and wrong line number, but if you click on them you can see the source mapped stack trace and jump into the correct file. I think this is ok for now, I'll continue to monitor the issue.
Please test and hopefully confirm my findings :)

Steps to test:
- Check out branch, introduce error in the code and load webknossos. Check the error message and whether you can jump into the source mapped file. Also check whether you can properly set breakpoints and debug.

Issues:
- fixes my sanity

------
